### DISCUSSION TITLE
error when tryin to run section-properties

### DISCUSSION THREAD

**Discussion by rlenner** [robbievanleeuwen/section-properties#407] [General]
Hi, 
apologies if this question is completely rudimentary, but I have trouble running the section properties. The pip install seem to work fine, but when I try run some basic examples I get the following: 

SystemError                               Traceback (most recent call last)
Cell In[3], line 1
----> 1 from sectionproperties.analysis import Section
      2 from sectionproperties.pre import CompoundGeometry

File ~\AppData\Local\anaconda3\lib\site-packages\sectionproperties\analysis\__init__.py:3
      1 """sectionproperties analysis module."""
----> 3 from sectionproperties.analysis.section import Section

File ~\AppData\Local\anaconda3\lib\site-packages\sectionproperties\analysis\section.py:27
     24 from shapely import Point, Polygon
     25 from shapely.strtree import STRtree
---> 27 import sectionproperties.analysis.fea as fea
     28 import sectionproperties.analysis.plastic_section as sp_plastic
     29 import sectionproperties.analysis.solver as solver

File ~\AppData\Local\anaconda3\lib\site-packages\sectionproperties\analysis\fea.py:24
     22 # numba is an optional dependency
     23 try:
---> 24     from numba import njit
     25 except ImportError:
     27     def njit(**options: Any) -> Callable[[Any], Any]:

File ~\AppData\Local\anaconda3\lib\site-packages\numba\__init__.py:42
     38 from numba.core.decorators import (cfunc, generated_jit, jit, njit, stencil,
     39                                    jit_module)
     41 # Re-export vectorize decorators and the thread layer querying function
---> 42 from numba.np.ufunc import (vectorize, guvectorize, threading_layer,
     43                             get_num_threads, set_num_threads,
     44                             set_parallel_chunksize, get_parallel_chunksize,
     45                             get_thread_id)
     47 # Re-export Numpy helpers
     48 from numba.np.numpy_support import carray, farray, from_dtype

File ~\AppData\Local\anaconda3\lib\site-packages\numba\np\ufunc\__init__.py:3
      1 # -*- coding: utf-8 -*-
----> 3 from numba.np.ufunc.decorators import Vectorize, GUVectorize, vectorize, guvectorize
      4 from numba.np.ufunc._internal import PyUFunc_None, PyUFunc_Zero, PyUFunc_One
      5 from numba.np.ufunc import _internal, array_exprs

File ~\AppData\Local\anaconda3\lib\site-packages\numba\np\ufunc\decorators.py:3
      1 import inspect
----> 3 from numba.np.ufunc import _internal
      4 from numba.np.ufunc.parallel import ParallelUFuncBuilder, ParallelGUFuncBuilder
      6 from numba.core.registry import DelayedRegistry

SystemError: initialization of _internal failed without raising an exception

**Comment by rlenner** [comment]
**from the installation **


(base) C:\Users\roman_lenner_jr>pip install sectionproperties
Requirement already satisfied: sectionproperties in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (3.2.0)
Requirement already satisfied: shapely<3.0.0,>=2.0.3 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from sectionproperties) (2.0.3)
Requirement already satisfied: more-itertools<11.0.0,>=10.2.0 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from sectionproperties) (10.2.0)
Requirement already satisfied: rich<14.0.0,>=13.7.1 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from sectionproperties) (13.7.1)
Requirement already satisfied: matplotlib<4.0.0,>=3.8.3 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from sectionproperties) (3.8.3)
Requirement already satisfied: click<9.0.0,>=8.1.7 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from sectionproperties) (8.1.7)
Requirement already satisfied: numpy<2.0.0,>=1.26.4 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from sectionproperties) (1.26.4)
Requirement already satisfied: triangle<20230924,>=20230923 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from sectionproperties) (20230923)
Requirement already satisfied: scipy<2.0.0,>=1.12.0 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from sectionproperties) (1.12.0)
Requirement already satisfied: colorama in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from click<9.0.0,>=8.1.7->sectionproperties) (0.4.6)
Requirement already satisfied: fonttools>=4.22.0 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from matplotlib<4.0.0,>=3.8.3->sectionproperties) (4.25.0)
Requirement already satisfied: packaging>=20.0 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from matplotlib<4.0.0,>=3.8.3->sectionproperties) (22.0)
Requirement already satisfied: pillow>=8 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from matplotlib<4.0.0,>=3.8.3->sectionproperties) (9.4.0)
Requirement already satisfied: kiwisolver>=1.3.1 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from matplotlib<4.0.0,>=3.8.3->sectionproperties) (1.4.4)
Requirement already satisfied: contourpy>=1.0.1 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from matplotlib<4.0.0,>=3.8.3->sectionproperties) (1.0.5)
Requirement already satisfied: pyparsing>=2.3.1 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from matplotlib<4.0.0,>=3.8.3->sectionproperties) (3.0.9)
Requirement already satisfied: cycler>=0.10 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from matplotlib<4.0.0,>=3.8.3->sectionproperties) (0.11.0)
Requirement already satisfied: python-dateutil>=2.7 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from matplotlib<4.0.0,>=3.8.3->sectionproperties) (2.8.2)
Requirement already satisfied: markdown-it-py>=2.2.0 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from rich<14.0.0,>=13.7.1->sectionproperties) (3.0.0)
Requirement already satisfied: pygments<3.0.0,>=2.13.0 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from rich<14.0.0,>=13.7.1->sectionproperties) (2.17.2)
Requirement already satisfied: mdurl~=0.1 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from markdown-it-py>=2.2.0->rich<14.0.0,>=13.7.1->sectionproperties) (0.1.2)
Requirement already satisfied: six>=1.5 in c:\users\roman_lenner_jr\appdata\local\anaconda3\lib\site-packages (from python-dateutil>=2.7->matplotlib<4.0.0,>=3.8.3->sectionproperties) (1.16.0)

**Comment by robbievanleeuwen** [comment]
Gday @rlenner, it looks like there is an issue with your `numba` installation. This isn't installed by default when you `pip install sectionproperties`, so an incompatible version might be hanging out in your virtual environment?

I would recommend installing `sectionproperties` in a fresh virtual environment. It looks like you are using anaconda - I think these are the steps you need to follow to achieve this:

```bash
conda create --name sp python=3.12
conda activate sp
pip install sectionproperties
```

You can choose any name you'd like, I've just chosen `sp`, you can also chose python 3.9 through to 3.12. Make sure you activate this virtual environment every time you want to work with `sectionproperties`. There's a little anaconda cheatsheet [here](https://static.anaconda.cloud/content/b9d8b9644a1a9bcfad3fc22d20b76f08e45a9cf759a6ab792040a3e6e2409174).

If you do want to use `numba` as well (can provide some non-trivial efficiencies with large meshes), you can replace the last line with `pip install sectionproperties[numba]`. However some users have had trouble running `numba` on windows, #347. More info on installation can be found [here](https://sectionproperties.readthedocs.io/en/stable/installation.html).

Hope this helps!

**Reply by robbievanleeuwen** [reply]
As a follow up, it looks like `numba` is installed by default in a new anaconda environment. You could try uninstalling with `conda uninstall numba --name sp`.

**Comment by rlenner** [comment]
Hi Robbie, 

perfect, all solved running the in sp environment 

Cheers 
Roman
